### ISSUE TITLE
error when using a POSIXt object for dateField

### ISSUE THREAD

**Issue by tomjemmett** [nhs-r-community/NHSRplotthedots#56]
the plot method works fine if we pass a "Date" data type, e.g.
``` r
tibble(x = Sys.Date() + 1:20, y = rnorm(20)) %>% spc("y", "x")
```

however, if you are using a POSIXt object (e.g. a datetime), you will get an error with ggplot
``` r
> tibble(x = Sys.now() + 1:20, y = rnorm(20)) %>% spc("y", "x")
```

that's because you have to choose between `scale_x_date` or `scale_x_datetime`.

options to solve this would be to detect type of column used and use an appropriate scale_x function, or always convert from Date -> POSIXt

**Comment by ThomUK** [comment]
I'm glad you raised this issue @tomjemmett. I've had need to run SPC on twice daily data before (8am shift and 8pm shift), so moving to datetime would make that possible in this tool.

It would be great to have that option (even if the majority of user won't need it (ie. daily, weekly, or monthly data)).

**Comment by tomjemmett** [comment]
I've pushed a fix in #48 that handles this by first checking the data type is either Date or POSIXt (so should handle both posix types). This could cause some edge cases though... I have to choose a timezone so force everything to UTC... it *may* be better to more selectively change the date column (if it's Date, promote to POSIXct)
